extension/observer/hostobserver/extension.go

```diff
@@ -185,9 +193,10 @@ type connectionDetails struct {
 
 func collectConnectionDetails(c *net.ConnectionStat) connectionDetails {
 	ip := c.Laddr.IP
-	// An IP addr of 0.0.0.0 means it listens on all interfaces, including
-	// localhost, so use that since we can't actually connect to 0.0.0.0.
-	if ip == "0.0.0.0" {
+	// An IP addr of 0.0.0.0 (or "*" on darwin) means it listens on all
+	// interfaces, including localhost, so use that since we can't
+	// actually connect to 0.0.0.0.
+	if ip == "0.0.0.0" || ip == "*" {
```

Comment:
Am I correct in seeing this as the bug and the added test is to actually require runs on mac per `getExpectedHost`?  If so, why not (also) update `TestCollectConnectionDetails` to include an additional mocked item to remove the platform requirement?

Reply:
Yes, that was the bug on Mac. Added an additional test case in `TestCollectConnectionDetails` as well.